apps/designer/app/designer/features/props-panel/use-props-logic.ts

```diff
@@ -19,11 +19,16 @@ declare module "~/shared/pubsub" {
   }
 }
 
-export type handleChangePropType = (
-  id: UserProp["id"],
-  field: "prop" | "value",
-  value: UserProp["prop"] | UserProp["value"]
-) => void;
+// First infer is used to iterate over Union type
+type UserPropValue = UserProp extends infer IterUserProp
+  ? IterUserProp extends { value: infer Value }
+    ? Value
+    : never
+  : never;
```

Comment:
almost too complex type for my test, please check if it can be simplified some how